### DISCUSSION TITLE
Work started ... Moving to Version 3

### DISCUSSION THREAD

**Discussion by fboucher** [microsoft/AzUrlShortener#296] [General]
I'm thinking to make some important changes to the Ulr Shortener and because it will have a lot of impacts this is why I think it required a major version increment.

👉The goal is to make this tool easy to get started without having to worry about spending too much. 

Version 3 will:
- will be 100% targeting the Azure Static Web App (now GA)
- security will be simplify and will support (GitHub, Google, Azure AD, Twitter, and Facebook)
- Instead of have the code in 2 repository all the code will be in this repo
- A roadmap of the prioritized features will be available
- Big cleanup of the branches, and all pages will be done
- V1 (still used by many users) will be moved in a separated repo.

**Any thoughts, or suggestions?**

**Would you like to help?**

**Comment by fboucher** [comment]
The work for V3 is starting. You can follow the work on branch dev-v3

**Reply by dmitriilezine** [reply]
hello, can you share potential release timeframe for v3, all in one type of deployment?

**Reply by fboucher** [reply]
Sorry I missed this message.:(
I want version 3 to be out before the end of summer.
THe AzUrlShortener (the redirect) tool is mostly finished.
- Code is done
- Documentation is partially completed
- deployment will be a 1-click deployment. Then a step by step guide to put a custom domain.

I can be use just like this but creating new URL using Azure Storage Explorer is not great.

This is why I'm also working on upgrading the TinyBlazorAdmin. Code is progressing well. I'm planning to have a 1-click deployment for this project too. With a few customizable step after, like adding users.
